src/core/DataFile.cpp

```diff
@@ -779,10 +792,38 @@ void DataFile::loadData( const QByteArray & _data, const QString & _sourceFile )
 	m_type = type( root.attribute( "type" ) );
 	m_head = root.elementsByTagName( "head" ).item( 0 ).toElement();
 
-	if( root.hasAttribute( "creatorversion" ) &&
-		root.attribute( "creatorversion" ) != LMMS_VERSION )
+
+	if( root.hasAttribute( "creatorversion" ) )
 	{
-		upgrade();
+		//compareType defaults to Build,so it doesn't have to be set here
+		ProjectVersion createdWith = root.attribute( "creatorversion" );
+		ProjectVersion openedWith = LMMS_VERSION;;
+
+		if (createdWith != openedWith)
+		{
+			//Only one compareType needs to be set, and "[The] ProjectVersion return type
+			//from the setCompareType(...) function [...] saves a few lines of code!" (@tresf)
```

Comment:
this comment should rewritten as easier to read.  It's a bit confusing with all of my quoted text in there... :)

FYI.